src/html-imports.js

```diff
@@ -9,718 +9,754 @@
-      at.value = (a === 'style') ?
-        path.resolveUrlsInCssText(v, base) :
-        path.replaceAttrUrl(v, base);
+  function fixUrls(element, base) {
+    const n$ = element.querySelectorAll(importsSelectors);
+    for (let i = 0, l = n$.length, n; i < l && (n = n$[i]); i++) {
+      // Ensure we add load/error listeners before modifying urls or appending
+      // these to the main document.
+      whenElementLoaded(n);
+      if (n.href) {
+        n.setAttribute('href', Path.replaceAttrUrl(n.getAttribute('href'), base));
+      }
+      if (n.src) {
+        n.setAttribute('src', Path.replaceAttrUrl(n.getAttribute('src'), base));
+      }
+      if (n.localName == 'style') {
+        Path.resolveUrlsInStyle(n, base);
+      } else if (n.localName === 'script') {
+        if (n.textContent) {
+          n.textContent += `\n//# sourceURL=${base}`;
+        }
+        // NOTE: we override the type here, might need to keep track of original
+        // type and apply it to clone when running the script.
+        n.setAttribute('type', scriptType);
+      }
     }
+    fixUrlsInTemplates(element, base);
   }
-}
-
-function fixUrlsInTemplate(template, base) {
-  var content = template.content;
-  var n$ = content.querySelectorAll('style, form[action], [src], [href], [url], [style]');
-  for (var i=0; i < n$.length; i++) {
-    var n = n$[i];
-    if (n.localName == 'style') {
-      path.resolveUrlsInStyle(n, base);
-    } else {
-      fixUrlAttributes(n, base);
+
+  function fixDomModules(element, url) {
+    const s$ = element.querySelectorAll('dom-module');
+    for (let i = 0; i < s$.length; i++) {
+      const o = s$[i];
+      const assetpath = o.getAttribute('assetpath') || '';
+      o.setAttribute('assetpath', Path.replaceAttrUrl(assetpath, url));
     }
   }
-  fixUrlsInTemplates(content, base);
-}
 
-function fixUrlsInTemplates(element, base) {
-  var t$ = element.querySelectorAll('template');
-  for (var i=0; i < t$.length; i++) {
-    fixUrlsInTemplate(t$[i], base);
+  /**
+   * Replaces all the imported scripts with a clone in order to execute them.
+   * Updates the `currentScript`.
+   * @param {!HTMLDocument} doc
+   * @return {Promise} Resolved when scripts are loaded.
+   */
+  function runScripts(doc) {
+    const s$ = doc.querySelectorAll(`script[type=${scriptType}]`);
+    let promise = Promise.resolve();
+    for (let i = 0, l = s$.length, s; i < l && (s = s$[i]); i++) {
+      promise = promise.then(() => {
+        const c = doc.createElement('script');
+        c.textContent = s.textContent;
+        if (s.src) {
+          c.setAttribute('src', s.getAttribute('src'));
+        }
+        // Listen for load/error events before adding the clone to the document.
+        // Catch failures, always return c.
+        const whenLoadedPromise = whenElementLoaded(c).catch(() => c);
+        // Update currentScript and replace original with clone script.
+        currentScript = c;
+        s.parentNode.replaceChild(c, s);
```

Comment:
Let's add a comment somewhere (either here or in `whenElementLoaded`) that you're relying on `replaceChild` of an inline script to run the script synchronously before returning, since `whenElementLoaded` doesn't wait for the load event since there is no `src`.